### DISCUSSION TITLE
How to use Custom runtimeconfig

### DISCUSSION THREAD

**Discussion by divyam234** [DuCanhGH/next-pwa#99] [General]
I want to fully override runtimeConfig instead of extending like next-pwa

**Comment by DuCanhGH** [comment]
Just set `extendDefaultRuntimeCaching` to `false`. I thought this option is disabled by default?

**Comment by flyingwolf1701** [comment]
I have a question about this as well. 

I created a file called runtimeCachingConfig will this overwrite cache.js? or is the a better way to use my own file instead of cache.js?

const withPWA = require('@ducanh2912/next-pwa').default({
  dest: 'public',
  register: true,
  extendDefaultRuntimeCaching: true,
  maximumFileSizeToCacheInBytes: 10 * 1024 * 1024, // Set limit to 10 MB
  workboxOptions: {
    runtimeCaching: changeRuntimeCaching,
  },
  debug: true,
});

I need to change almost all the cache.js settings. And I can't figure out a better way than this. but I don't think its working

**Comment by DuCanhGH** [comment]
@flyingwolf1701 we don't read whatever is `runtimeCachingConfig`. There's only one way to customize `runtimeCaching`, which is `workboxOptions.runtimeCaching`. Also, if you are going to overwrite `cache.js` set `extendDefaultRuntimeCaching` to `false`. And by the way, `maximumFileSizeToCacheBytes` and `debug` are not valid `next-pwa` options, move the former to `workboxOptions.maximumFileSizeToCacheBytes` and remove the latter. You may want to add `// @ts-check` to the top of your `next.config.js` to get hints from your IDE whenever you provide an invalid option.

**Reply by flyingwolf1701** [reply]
Thank you so much! And thank you for picking up this project and running with it, you are making a great contribution to the dev community!
